code/datums/atom_hud.dm

```diff
@@ -101,6 +103,8 @@ GLOBAL_LIST_INIT(huds, list(
 	if(!A || (A in hudatoms))
 		return FALSE
 	hudatoms |= A
+	if(istype(A, /obj/effect/temp_visual))
+		RegisterSignal(A, COMSIG_PARENT_QDELETING, .proc/remove_from_hud)
```

Comment:
i'm afraid this might lead one day to double registering runtime. Nothing in test so far

Reply:
can this proc be called twice without the unregistering one ?

Reply:
i don't know. I don't think so, it shouldn't be, but this code is used by a lot and a lot of things